src/metabase/sync/sync_metadata/fields.clj

```diff
@@ -137,29 +157,42 @@
   "Find Metadata that matches FIELD-METADATA from a set of OTHER-METADATA, if any exists."
   [field-metadata :- TableMetadataFieldWithOptionalID, other-metadata :- #{TableMetadataFieldWithOptionalID}]
   (some (fn [other-field-metadata]
-          (when (= (str/lower-case (:name field-metadata))
-                   (str/lower-case (:name other-field-metadata)))
+          (when (= (canonical-name field-metadata)
+                   (canonical-name other-field-metadata))
               other-field-metadata))
         other-metadata))
 
+(declare sync-field-instances!)
+
+(s/defn ^:private update-field-chunk!
+  [table :- i/TableInstance, known-fields :- {s/Str TableMetadataFieldWithID}, fields-to-update :- [i/TableMetadataField]]
+  (doseq [our-field fields-to-update
+          :let [db-field (get known-fields (canonical-name our-field))]]
+    ;; if field exists in both metadata sets then make sure the data recorded about the Field such as database_type is
+    ;; up-to-date...
+    (update-field-metadata-if-needed! table db-field our-field)
+    ;; ...and then recursively check the nested fields.
+    (when-let [db-nested-fields (seq (:nested-fields our-field))]
+      (sync-field-instances! table (set db-nested-fields) (:nested-fields db-field) (:id our-field)))))
+
 (s/defn ^:private sync-field-instances!
   "Make sure the instances of Metabase Field are in-sync with the DB-METADATA."
   [table :- i/TableInstance, db-metadata :- #{i/TableMetadataField}, our-metadata :- #{TableMetadataFieldWithID}
    parent-id :- ParentID]
-  ;; Loop thru fields in DB-METADATA. Create/reactivate any fields that don't exist in OUR-METADATA.
-  (doseq [db-field db-metadata]
-    (sync-util/with-error-handling (format "Error checking if Field '%s' needs to be created or reactivated"
-                                           (:name db-field))
-      (if-let [our-field (matching-field-metadata db-field our-metadata)]
-        ;; if field exists in both metadata sets then...
-        (do
-          ;; ...make sure the data recorded about the Field such as database_type is up-to-date...
-          (update-field-metadata-if-needed! table our-field db-field)
-          ;; ...and then recursively check the nested fields.
-          (when-let [db-nested-fields (seq (:nested-fields db-field))]
-            (sync-field-instances! table (set db-nested-fields) (:nested-fields our-field) (:id our-field))))
-        ;; otherwise if field doesn't exist, create or reactivate it
-        (create-or-reactivate-field! table db-field parent-id))))
+  (let [known-fields (u/key-by canonical-name our-metadata)]
+    ;; Loop thru fields in DB-METADATA. Create/reactivate any fields that don't exist in OUR-METADATA.
+    (doseq [db-field-chunk (partition-all 1000 db-metadata)]
+      (sync-util/with-error-handling (format "Error checking if Fields '%s' needs to be created or reactivated"
+                                             (map :name db-field-chunk))
```

Comment:
I need to `pr-str` the `db-field-chunk`. This is showing as `clojure.lang.LazySeq@7860` in error messages right now.